src/marks/axis.d.ts

```diff
@@ -33,9 +33,9 @@ export interface AxisXOptions extends AxisOptions, TickXOptions {}
 
 export interface AxisYOptions extends AxisOptions, TickYOptions {}
 
-export interface GridXOptions extends GridOptions, RuleXOptions {}
+export interface GridXOptions extends GridOptions, Omit<RuleXOptions, "interval"> {}
```

Comment:
This is needed because the rule mark allows a base (non-range) interval, whereas the grid mark requires a range interval. You can widen a type when extending, but you can’t narrow it.